lucetc/src/module.rs

```diff
@@ -106,7 +111,22 @@ impl<'a> ModuleEnvironment<'a> for ModuleInfo<'a> {
             0,
             ir::AbiParam::special(NATIVE_POINTER, ir::ArgumentPurpose::VMContext),
         );
-        self.signatures.push(sig);
+
+        let mut maybe_match_key: Option<UniqueSignatureIndex> = None;
+
+        for (k, v) in self.signatures.iter() {
+            if v == &sig {
+                maybe_match_key = Some(k);
+            }
+        }
```

Comment:
I would use `find` for this, its short circuiting (stops iterating after a match) and shorter: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.find

Reply:
`find` is definitely better (and is what i wanted - i thought it wasn't there for some reason)